widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTabPanel.java

```diff
@@ -32,13 +34,15 @@ public PentahoTabPanel() {
     setStylePrimaryName( "pentaho-tab-panel" );
     tabBar.setStylePrimaryName( "pentaho-tab-bar" );
     add( tabBar );
+    Roles.getTablistRole().set( tabBar.getElement() );
     add( tabDeck );
     tabDeck.getElement().getParentElement().setClassName( "pentaho-tab-deck-panel" );
     tabDeck.setHeight( "100%" );
   }
 
   public void addTab( String text, String tooltip, boolean closeable, Widget content ) {
     PentahoTab tab = new PentahoTab( text, tooltip, this, content, closeable );
+    tab.addStyleName( "focus-visible" );
```

Comment:
Imo, this is an internal detail of the `PentahoTab` class itself and should be moved into it.

Reply:
Hi @dcleao. from the comments added for PentahoTab class and PentahoTabPanel class whatever changes you are suggesting are not feasible. PentahoTabPanel is the entire row of tabs and PentahoTab is each tab inside it.

we have to use listeners, tabindex, focus on each PentahoTab not the panel. And focus as you mentioned an option is a must not an option. you can refer here https://www.w3.org/WAI/ARIA/apg/example-index/tabs/tabs-automatic.html
and https://www.w3.org/WAI/ARIA/apg/example-index/tabs/js/tabs-automatic.js

_tab.addEventListener('keydown', this.onKeydown.bind(this));_

the EventListener is added to the Tab right ?

_tab.focus();_

the focus is set on Tab right ?

I am not sure what are we going to achieve out of implementing Focusable ? as the code is already working and unit tested as per the standard. Implementing Focusable is an option, but is it must ? really ? needed ???